rclcpp/src/rclcpp/experimental/executors/events_executor/events_executor.cpp

```diff
@@ -341,6 +341,10 @@ EventsExecutor::refresh_current_collection(
   // Acquire lock before modifying the current collection
   std::lock_guard<std::mutex> guard(mutex_);
 
+  // Remove expired timers before updating the timers collection
```

Comment:
@apojomovsky did we verify whether this issue was specific to timers?
This piece of is not timer-specific, so maybe the comment shouldn't mention timers and it should just point to this PR.

Reply:
I did some experiments with variations of the approach used for the timers and I couldn't reproduce it for the rest of the entities. This said, my knowledge is limited so I cannot ensure it.
I updated the comment according to your suggestions.